streaming/src/main/scala/org/apache/spark/streaming/StreamingContext.scala

```diff
@@ -607,8 +614,27 @@ class StreamingContext private[streaming] (
  */
 
 object StreamingContext extends Logging {
+  /**
+   * Lock that guards access to global variables that track active StreamingContext.
+   */
+  private val ACTIVATION_LOCK = new Object()
 
-  private[streaming] val DEFAULT_CLEANER_TTL = 3600
+  private var activeContext: AtomicReference[StreamingContext] =
+    new AtomicReference[StreamingContext](null)
+
+  private def assertNoOtherContextIsActive(): Unit = {
+    ACTIVATION_LOCK.synchronized {
+      if (activeContext.get() != null) {
+        throw new SparkException("Only one StreamingContext may be started in this JVM")
```

Comment:
To clarify, we're allowed to start multiple StreamingContexts provided that only one context is running at a time, right?  Since that's the case, what do you think about rewording this to something like "Only one StreamingContext may be running in this JVM"?  I just worry that "started" might be misinterpreted as saying that you can only _create_ one SparkContext per JVM.

We don't need to add an explicit test for starting a new StreamingContext after stopping the old one since this case is implicitly handled by all of our other test suites.